explainer.md

```diff
@@ -0,0 +1,369 @@
+  }).catch(function (err) {
+    // No acceptable authenticator or user refused consent. Handle appropriately.
+  });
+```
+
+As per the above note, the need to have a special enrollment flow for SPC will
+depend on whether:
+
+* [WebAuthn] starts to allow credential creation in a cross-origin iframe.
+* If it is considered reasonable for 'vanilla' WebAuthn credentials to be used
+   in SPC authentication.
+
+If both of these become true, then the extension would not be needed during
+enrollment.
+
+##### Creating a SPC credential in a cross-origin iframe
+
+Unlike normal WebAuthn credentials, SPC allows a credential to be created in a
+cross-origin iframe (e.g. if `merchant.com` embeds an iframe from `bank.com`). This is
+intended to support the common enrollment flow of a bank enrolling the user during a
+step-up challenge (e.g. after proving their identity via OTP).
+
+To allow this, the cross-origin iframe must have the ["payment" permission
+policy] set. For example:
+
+```html
+<!-- Assume parent origin is merchant.com -->
+<!-- Inside this cross-origin iframe, script would be allowed to create a SPC credential for example.org -->
+<iframe src="https://example.org" allow="payment">
+```
+
+#### Authenticating a payment
+
+An origin may invoke the [Payment Request API] with the
+`secure-payment-confirmation` payment method to prompt the user to verify a
+Secure Payment Confirmation credential created by any other origin. The User
+Agent will display a native user interface with transaction details (e.g. the
+payment amount and the payee origin).
+
+> **NOTE**: The `PaymentRequest.show()` method requires a user gesture. The User
+> Agent will display a native user interface with the payment amount and the
+> payee origin.
+
+> **NOTE**: [Per the Payment Request specification][pr-cross-origin], if
+> `PaymentRequest` is used within a cross-origin iframe (e.g. if `merchant.com`
+> embeds an iframe from `psp.com`, and `psp.com` wishes to use
+> `PaymentRequest`), that iframe must have the ["payment" permission policy]
+> set.
+
+Proposed new `secure-payment-confirmation` payment method:
+
+```javascript
+const request = new PaymentRequest([{
+  supportedMethods: "secure-payment-confirmation",
+  data: {
+    // List of credential IDs obtained from the ???.
+    credentialIds,
+
+    // The challenge is also obtained from the ???.
+    challenge: new Uint8Array(
+        randomStringFromServer, c => c.charCodeAt(0)),
+
+    instrument: {
+      displayName: "Fancy Card ****1234",
+      icon: "https://fancybank.com/card-art.png",
+    },
+
+    payeeOrigin: "https://merchant.com",
+
+    timeout: 60000,  // 1 minute
+  }], {
+    total: {
+      label: "Total",
+      amount: {
+        currency: "USD",
+        value: "5.00",
+      },
+    },
+  });
+
+try {
+  const canMakePayment = await request.canMakePayment();
+  if (!canMakePayment) { throw new Error('Cannot make payment'); }
```

Comment:
Please mention that both `canMakePayment()` and `hasEnrolledInstrument()` check only that a user verifying platform authenticator is available. To preserve user privacy, they do not check whether any credentials are on file.